### ISSUE TITLE
macOS menu too small and red flickering

### ISSUE THREAD

**Issue by StenApp** [ioquake/ioq3#351]
![bildschirmfoto 2018-02-25 um 16 23 19](https://user-images.githubusercontent.com/32524257/36643164-6afb387e-1a48-11e8-966b-3b45bc2751f3.png)
New issue:
Resolutions greater 1024x768 lead to the menu being to small (800x600?) on the left down side and the rest ist filled with red flickering.

**Comment by TimeDoctor** [comment]
Can you tell us which model of Mac you have, OS version, what graphics chipset, and what display it uses? Also, have you set the resolution to your desktop resolution?

**Comment by acheronfail** [comment]
This happens on my MacBook Pro (15-inch, 2017).

This has an inbuilt graphics chipset: Intel HD Graphics 630 1536 MB
And an dedicated one: Radeon Pro 555

This happens on most resolutions, except 800x600 and 1024x768.

**Comment by ensiform** [comment]
Can you get a console log? Using \condump command. It will appear in the directory where your config file is located.

I think trying to play quake3 and other older GL games on Mac is one going to get harder and harder as Apple decides to be anti OpenGL and anti old software more aggressively.

**Comment by acheronfail** [comment]
> I think trying to play quake3 and other older GL games on Mac is only going to get harder and harder as Apple decides to be anti OpenGL and anti old software more aggressively.

Too true. 😢 

Here's the output of the `\condump` command:

[log.txt](https://github.com/ioquake/ioq3/files/2892362/log.txt)

**Comment by acheronfail** [comment]
I've noticed in the log that the available resolutions don't match those that can be chosen in Q3's settings, is that supposed to happen?

in log.txt:
```
Available modes: '1152x720 1280x800 1440x900 1680x1050 2048x1280 2560x1600
2880x1800 3360x2100 840x524 1650x1050 640x480 800x600 1024x768'
```

But in Q3's settings there are much fewer to choose from. I wonder if that's why we're getting this flickering (since the ratios aren't exactly the same).

**Comment by acheronfail** [comment]
FWIW these are the available resolutions in the Q3 settings menu:

```
856x480 wide screen 320x240 400x300 512x384 640x480 800x600 960x720 1024x768 1152x864 1280x1024 1600x1200 2048x1536
```

**Comment by acheronfail** [comment]
Yep, if I set the following:

```
seta r_customheight "1800"
seta r_customwidth "2880"
seta r_fullscreen "1"
seta r_mode "-1"
```

Then the resolution is correct and there's no red flicker.

Is there a way we can make sure the right values are available to choose from in the Q3 settings menu? 🤔

**Comment by ensiform** [comment]
No, the menu can only be fixed if you're running the ioq3 fixed "mod". If you play vq3 UI qvm or other mod you can't do anything. So this isn't a bug.

Alternatively try r_mode -2

**Comment by acheronfail** [comment]
> No, the menu can only be fixed if you're running the ioq3 fixed "mod". If you play vq3 UI qvm or other mod you can't do anything. So this isn't a bug.

Makes sense.

> Alternatively try r_mode -2

@ensiform siform how does `r_mode "-2"` differ from `r_mode "-1"`?

**Comment by ensiform** [comment]
-2 is desktop resolution, ioq3 specific.

**Comment by briancullinan** [comment]
Hello!! I just ran into this and it was so bizarre I had to investigate. I think what happened is recently there was a OSX change and someone updated for the XCode tool change. DELETE YOUR BUILD CACHE!

Delete the directories below build/* whenever you pull or update some build system code, or tooling changes out from under you, you must delete the .o files inside build, or just `rm ./build/*`

Very strange, build was the only thing that was different, and the red flashing went away and the resolution is normal again. Sad this has nothing to do with video settings, maybe there is a C compiler option to help with this virtually "pulling the rug out from under me" and my project :(.  Lost days of work because of sadness. Thank you again APPLE! They hate developers.

**Comment by StenApp** [comment]
> Can you tell us which model of Mac you have, OS version, what graphics chipset, and what display it uses? Also, have you set the resolution to your desktop resolution?

It's a MacBookPro 2017 15" with a Radeon Pro 560 with a 2880 x 1800 Retina display

**Comment by DataGreed** [comment]
Same thing on 2019 16"

**Comment by DataGreed** [comment]
> I've noticed in the log that the available resolutions don't match those that can be chosen in Q3's settings, is that supposed to happen?
> 
> in log.txt:
> 
> ```
> Available modes: '1152x720 1280x800 1440x900 1680x1050 2048x1280 2560x1600
> 2880x1800 3360x2100 840x524 1650x1050 640x480 800x600 1024x768'
> ```
> 
> But in Q3's settings there are much fewer to choose from. I wonder if that's why we're getting this flickering (since the ratios aren't exactly the same).

where can you find this log to see the available modes?

**Comment by NuclearMonster** [comment]
You will need to choose a resolution that matches your display, or `r_mode -2` as mentioned above. Please see our Player's Guide on ioquake3.org for help or continue discussing this issue in our discourse forums. 

I hope some day we can have a better solution with an external piece of software, some might say a launcher of sorts, that chooses better resolutions and supplants the menus where possible.
